topostats/roughness.py

```diff
@@ -0,0 +1,17 @@
+"""Contains functions pertaining to the roughness of AFM images."""
+
+import numpy as np
+
+
+def roughness_rms(image: np.ndarray) -> float:
+    """Calculate the rms roughness of a heightmap image.
+    Parameters
+    ----------
+    image: np.ndarray
+        2D numpy array of heightmap data to calculate the roughness of
+
+    Returns:
+    float
+        The RMS roughness of the input array.
+    """
+    return np.sqrt(np.mean(np.square(image)))
```

Comment:
Just to check this is the metric that is required?

I would have thought "roughness" might measure some element of how much variation there is across an image and so the something like the [Variance](https://en.wikipedia.org/wiki/Variance)  (or it's square root [Standard Deviation](https://en.wikipedia.org/wiki/Standard_deviation) which is the same units as the original heihgts) might be more appropriate as that is quantifying how much variation there is across the sample.

That said scans of biological samples I'd expect (and have seen) tend to be bimodal and such statistics don't provide useful information. Perhaps of more relevance to scans of materials though.

Reply:
So this root-mean-squared roughness calculation is aimed at non-bio samples. An academic I'm working with has assured me that this is what is needed, so I'm taking their word for it. I have checked 😄 👍